src/scenario_tests.rs

```diff
@@ -1270,6 +1270,52 @@ mod src_tests {
         );
     }
 
+    #[test]
+    fn parsec_check_get_node_to_relocate_uses_online_nodes_only() {
+        let relocated_info_1 = get_relocated_info(CANDIDATE_1_OLD, SECTION_INFO_1);
```

Comment:
This is quite big.
I see we are being thorough here, which make sense, not sure I would have been so much.

I think we can simplify some part to make test easier to read:

I think for this test we want all the age the same.
One thing though is I think we need all names different and 1000+age was not doing that.

```rust
        let relocated_info = get_relocated_info(CANDIDATE_2_OLD, SECTION_INFO_2);
        // Could be a const at the top.
        let all_possible_states = [
            State::RelocatingAgeIncrease,
            State::RelocatingHop,
            State::RelocatingBackOnline,,
            State::Relocated(relocated_info),
            State::WaitingCandidateInfo(relocated_info),
            State::WaitingProofing,
            State::Offline,
        ];
        let inner_action = all_possible_states.iter().enumerate().fold(
            INNER_ACTION_YOUNG_ELDERS.clone(),
            |inner_action, (state, name_increment)| {
                let age = 5
                let node_state_with_enough_work_units = NodeState {
                    state,
                    work_units_done: age,
                    ..NodeState::default()
                };
                let node = Node(Attributes {
                    name: 1000 + name_increment,
                    age,
                });
                inner_action.extend_current_nodes_with(&node_state_with_enough_work_units, &[node])
            },
        );
        let initial_state = MemberState {
            action: Action::new(inner_action),
            ..Default::default()
        };
```


But Probably better to fix InnerAction missing interface:
```rust

// Add new
pub fn extend_current_nodes_states(self, nodes: &[(NodeState,Node)]) -> Self;



        let all_possible_states = [
            State::RelocatingAgeIncrease,
            State::RelocatingHop,
            State::RelocatingBackOnline,,
            State::Relocated(relocated_info),
            State::WaitingCandidateInfo(get_relocated_info(CANDIDATE_2_OLD, SECTION_INFO_2)),
            State::WaitingProofing,
            State::Offline,
        ];

        let age = 5;
        let all_possible_node_states_with_enough_work_unit = all_possible_states.iter().enumerate().map(
            |(state, name_increment)| {
                (NodeState {
                    state,
                    work_units_done: age,
                    ..NodeState::default()
                },
                Node(Attributes {
                    name: 1000 + name_increment,
                    age,
                }))
            },
        ).collect_vec();

        let initial_state = MemberState {
            action: Action::new(
                INNER_ACTION_YOUNG_ELDERS.clone().extend_current_nodes_states(all_possible_node_states_with_enough_work_unit)
            ),
            ..Default::default()
        };

        run_test(
            "Check get_node_to_relocate doesn't choose anything but 'Online' nodes",
            &initial_state,
            &[ParsecVote::WorkUnitIncrement.to_event()],
            &AssertState {
                action_our_events: vec![ActionTriggered::WorkUnitIncremented.to_event()],
            },
        );
    }
```

Reply:
I did mean to increment the age for each different name originally when I wrote `1000+age`, but the `relocated_info` made me change my plan and I forgot to update this part.  Good catch :)

I don't think it would break anything here, but I changed my plans when I wanted to keep the age consistent between the two candidates in the `relocated_info`s and their corresponding `Node`s.  I'll try to come up with a relatively clean approach to accommodate your suggestion but also keep these ages consistent.

Regarding the suggestion to add `extend_current_nodes_states()` (which I think is what @octol is hinting at too), I feel that adding a "helper" like this doesn't really help where it's doing trivial work and is only used in one instance.  However, as you're both on the same page, can I suggest just making the existing `extend_current_nodes()` public since that's what we want here, I think?  (There's no need to pass a tuple since `Node` is already part of `NodeState`).  Maybe I could add an assertion in that method to ensure we did actually add the expected number of new entries.  Something like that would have caught my mistake you spotted above.